include/Dialects/LinalgTransform/LinalgTransformOps.td

```diff
@@ -166,6 +166,44 @@ def LowerToLLVMOp : Linalg_Transform_Operation<"lower_to_llvm"> {
   let assemblyFormat = "attr-dict";
 }
 
+def GetParentLoopOp : Linalg_Transform_Operation<"get_parent_loop"> {
+  let description = [{Obtains a handle to a parent loop of the given
+  operation.}];
+
+  let arguments = (ins PDL_Operation:$target,
+                   DefaultValuedAttr<Confined<I64Attr, [IntPositive]>,
+                                     "1">:$num_loops);
+  let results = (outs PDL_Operation:$transformed);
+
+  let assemblyFormat = "$target attr-dict";
+}
+
+def UnrollLoopOp : Linalg_Transform_Operation<"unroll_loop"> {
+  let description = [{Unrolls the given loop with the given unroll factor.}];
+
+  let arguments = (ins PDL_Operation:$target,
+                   Confined<I64Attr, [IntPositive]>:$factor);
+  
+  let assemblyFormat = "$target attr-dict";
+}
+
+def PipelineLoopOp : Linalg_Transform_Operation<"pipeline_loop"> {
+  let arguments = (ins PDL_Operation:$target,
+                   DefaultValuedAttr<I64Attr, "1">:$modulo,
+                   DefaultValuedAttr<I64Attr, "10">:$read_latency);
+  let results = (outs PDL_Operation:$transformed);
+
+  let assemblyFormat = "$target attr-dict";
+}
+
+def OutlineLoopOp : Linalg_Transform_Operation<"outline_loop"> {
+  let arguments = (ins PDL_Operation:$target,
+                   StrAttr:$func_name);
+  let results = (outs PDL_Operation:$transformed);
+
+  let assemblyFormat = "$target attr-dict";
+}
+
```

Comment:
Out of curiosity, is there also a ready-made "FuseLoopsOp"? As in:
```
for i = 0:10:
for j=0:10:
```
becomes 
```
for ij= 0:100
```

Reply:
Not at the moment, this isn't something we needed so far, but can be added. The transformation is called loop coalescing btw.